src/utils/helpers/structure/addEvaluatedResponsesToStructure.ts

```diff
@@ -65,8 +70,11 @@ const addEvaluatedResponsesToStructure = async ({
     evaluationOptions
   )
   results.forEach((evaluatedElement, index) => {
-    flattenedElements[index].element = evaluatedElement
-    flattenedElements[index].response = responseObject[evaluatedElement.code]
+    const flattenedElement = flattenedElements[index]
+    flattenedElement.element = evaluatedElement
+    flattenedElement.response = responseObject[evaluatedElement.code]
+    if (flattenedElement.response)
+      flattenedElement.response.reviewResponse = reviewResponses[flattenedElement.element.id]
```

Comment:
I didn't understood why you are setting an reviewResponse inside the response here? The type `ResponseFull` in the state doesn't have a `review`. Are you already doing the change to use the new structure envisioned in [LOQ Structure Diagram](https://app.diagrams.net/#G1gzlvn8D9ulQP8baqo93qspjsQqEDlxOa)? If that's the case then I suggest copying the state to a new `PageStateNEW`, rename the field to `latestApplicationResponse` and also add the type to consider new fields? I think that would be clearer.

Reply:
Kind of need to link response to reviewResponse to know the progress (i.e. in the review progress PR I check against response having reviewResponse for assigned elements).

not 100% sure what you mean, but added changes to types. it looks like i didn't notice any type errors because previous flattenStructureElements function was returning `any[]`

9e6db37b5002f55a4d9a2238e40f9f298d56c885